packages/wp-source/src/libraries/handlers/author.ts

```diff
@@ -23,13 +23,13 @@ const authorHandler: Handler = async ({ route, params, state, libraries }) => {
     }
   });
 
-  // 3. throw an error if page is out of range
+  // 3. populate response and add page to data
+  const items = await populate({ response, state });
+  if (page > 0 && !items.length) throw new Error("Page doesn't exist.");
```

Comment:
I don't quite understand what this does. Could you please explain?

Reply:
If you fetch a page that is out of range, you get a response that could be an error message or an empty array. When that response is populated, `populate` returns an empty array because nothing was added to the state.

And if you access to the **first page** of an archive that exists but its empty, WP returns a page that says *Nothing Found* with code `200` meaning that archive has no content, but if you access the next page WP returns with code `404`.

So I check these two conditions before throwing an error.

Reply:
Oops, I just realized that if the author or taxonomy doesn't exist an error should be thrown. This is not happening right now.